SimulationAgent/SimulationRunner.cs

```diff
@@ -169,6 +167,9 @@ public void Start(Simulation simulation)
                 // of threads pending on the lock statement
                 this.starting = true;
 
+                // Override throttling limits in case they are provided by user
+                this.rateLimiting.SetCounters(simulation.RateLimits, this.log);
```

Comment:
Do we want to support changing these limits for a running simulation? If so, we'll need to figure out where these values need to be updated for a running simulation.

Reply:
For now, these are when we create a new Simulation as they are relative to a IoT Hub throttling limits.